src/Previewer/Internal.js

```diff
@@ -29,8 +30,16 @@ export default function Internal({
     builder.get_object("button_screenshot").visible = true;
   }
 
-  function open() {
-    if (!object_root) return;
+  async function open() {
+    if (!object_root) {
+      try {
+        await panel_ui.update();
```

Comment:
The flow is complicated for internal preview.

After the window is destroyed - `Internal` cannot rebuild it on its own and triggers an update on `panel_ui` which will call `internal.updateXML`. 

I wonder if instead – `Internal` shouldn't work like `External` and get a serializable string that it can feed into its own `Builder` whenever it needs.

Reply:
That seems reasonable. Making both previewers work in the same way would make maintainance easier, and I don't think it make it loose features. Only the GtkBuilder from the previewer would need to be handed over to the js code panel.